docker/compose/sawtooth-demo.yaml

```diff
@@ -18,52 +18,58 @@ version: "2.1"
 services:
 
   tp_config:
-    image: hyperledger/sawtooth-tp_config:unstable
+    image: hyperledger/sawtooth-tp_config:latest
     expose:
       - 40000
     depends_on:
       - validator
     entrypoint: tp_config -vv tcp://validator:40000
 
   tp_intkey_python:
-    image: hyperledger/sawtooth-tp_intkey_python:unstable
+    image: hyperledger/sawtooth-tp_intkey_python:latest
     expose:
       - 40000
     depends_on:
       - validator
     entrypoint: tp_intkey_python -vv tcp://validator:40000
 
   tp_xo_python:
-    image: hyperledger/sawtooth-tp_xo_python:unstable
+    image: hyperledger/sawtooth-tp_xo_python:latest
     expose:
       - 40000
     depends_on:
       - validator
     entrypoint: tp_xo_python -vv tcp://validator:40000
 
   validator:
-    image: hyperledger/sawtooth-validator:unstable
+    image: hyperledger/sawtooth-validator:latest
     expose:
       - 40000
+    ports:
+      - "40000:40000"
     # start the validator with an empty genesis batch
     entrypoint: "bash -c \"\
         sawtooth admin keygen && \
         sawtooth admin genesis && \
-        validator -vv\""
+        validator -vv \
+          --public-uri tcp://localhost:8800 \
```

Comment:
The --public-uri is where other validators can connect to this validator. Since this is a single validator, localhost will technically work, but it would probably be a good idea to use tcp://validator:8800 so that people don't get confused.